pkg/zkproofs/utils.go

```diff
@@ -0,0 +1,25 @@
+package zkproofs
+
+import (
+	"crypto/rand"
+	"errors"
+
+	"github.com/coinbase/kryptology/pkg/core/curves"
+)
+
+// Generates a non-zero random scalar. The chances of generating a zero scalar are very low.
+func GenerateRandomScalar(curve *curves.Curve) (curves.Scalar, error) {
```

Comment:
Also would be nice to have a test for this method. To be fully testable we have to have a good handle of the main collaborator which is `curve.Scalar.Random` in this case. I think for that we could have our own interface that returns a random scalar. We pass interface instead into the method. Real implementation still uses `curve.Scalar.Random` to generate scalars. At least one test implementation returns always zero to test for error case.

Reply:
Won't fix for now since this method is quite simple and the change required to make this testible is a bit weird. But agree that it would be good to have test coverage here eventually in case things change.

Reply:
Added a basic test for this method though